### ISSUE TITLE
invalid_object_definition occurs upon out-of-order data objects

### ISSUE THREAD

**Issue by dvpolitte** [NASA-PDS-Incubator/validate#614]
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When I use validate on a PDS4 product in which the File_Area_Observational contains multiple data objects but they are listed out-of-order, I receive errors of this form:

```
  FAIL: file:/Y:/daniel/validate_out_of_order_errors_test_case/ss__0505_0711794861_465rmo__0261222srlc10000w0__cgnj02.xml
      ERROR  [error.label.invalid_object_definition]   line 3: Invalid object definition for object #3.  The previously defined object ends at byte 473. Offset defined in label: 198
        1 product validation(s) completed
```

(In case it is agreed that this is a bug, the root of it is the use of [this function](https://github.com/NASA-PDS/validate/blob/7f76482dba67a28bcc5331be0daee3004019efb1/src/main/java/gov/nasa/pds/tools/validate/rule/pds4/DataDefinitionAndContentValidationRule.java#L130) that enforces this rule.)

### 🕵️ Expected behavior

I expected that there would be no error about this. I have searched for anything in the PDS4 standards documents that prohibit out-of-order tables, but I have not been able to find such a rule.

### 📜 To Reproduce

1. Extract the ZIP archive attached to this issue. It is a single Mars2020 SHERLOC product. Note that in its File_Area_Observational the third Table_Delimited has an offset less than the second's.
2. Run validate 3.1.1 or 3.0.3 with the command `validate ss__0505_0711794861_465rmo__0261222srlc10000w0__cgnj02.xml`
3. Observe the erorr message in the description above.

### 🖥 Environment Info

- Version of this software: 3.1.1
- Operating System: Windows 10 Pro 22H2

### 📚 Version of Software Used

```
PS> validate --version

gov.nasa.pds:validate
Version 3.1.1
Release Date: 2023-01-03 23:08:19

Copyright 2019, by the California Institute of Technology ("Caltech").
All rights reserved.
```

### 🩺 Test Data / Additional context

[validate_out_of_order_errors_test_case.zip](https://github.com/NASA-PDS/validate/files/11062303/validate_out_of_order_errors_test_case.zip)

### 🦄 Related requirements

None

### ⚙️ Engineering Details

Per DDWG meeting on 4/13, it was determined that data objects do not need to be in order within the label. We need to **order the objects by offset** prior to performing our validation steps.

**Comment by jordanpadams** [comment]
@dvpolitte this error is really just saying your object is defined incorrectly, isn't that expected since it is defined out of order?

The offset of object #3 is `198`, which is invalid because the tables defined before it are >198 bytes already.

are you saying the standards do not say things need to be defined in order in the label?

**Comment by dvpolitte** [comment]
@jordanpadams Yes, I'm saying that I cannot find where in the standards, if anywhere, that it is specified that the objects must be defined in order.

**Comment by jordanpadams** [comment]
@dvpolitte copy. I am asking around now. This may be something we bring to the DDWG for clarification.

**Comment by msbentley** [comment]
Just to chime in, this issue also hit me in recent days, and we've had to roll back to avoid it; since I see the fix pushed now, do you know what version this will land in? Just to be sure we skip to that one - thanks!

**Comment by jordanpadams** [comment]
@msbentley just tagged 17 hours ago:  https://github.com/NASA-PDS/validate/releases/tag/v3.2.0

**Comment by jordanpadams** [comment]
@dvpolitte as noted above, this bug fix should be in the latest version of validate. https://github.com/NASA-PDS/validate/releases/latest
